main.tf

```diff
@@ -0,0 +1,148 @@
+locals {
+  instance_count       = "${var.instance_enabled == "true" ? var.instance_count : 0}"
+  security_group_count = "${var.create_default_security_group == "true" ? 1 : 0}"
+  region               = "${var.region}"
+  root_iops            = "${var.root_volume_type == "io1" ? var.root_iops : "0"}"
+  ebs_iops             = "${var.ebs_volume_type == "io1" ? var.ebs_iops : "0"}"
+  availability_zone    = "${var.availability_zone}"
+  ami                  = "${var.ami != "" ? var.ami : data.aws_ami.default.image_id}"
+  root_volume_type     = "${var.root_volume_type != "" ? var.root_volume_type : data.aws_ami.info.root_device_type}"
+  count_default_ips    = "${var.associate_public_ip_address == "true" && var.assign_eip_address == "true" && var.instance_enabled == "true" && var.instance_count > 0 ? 1 : 0}"
+  ssh_key_pair_path    = "${var.ssh_key_pair_path == "" ? path.cwd : var.ssh_key_pair_path }"
+}
+
+data "aws_caller_identity" "default" {}
+
+data "aws_iam_policy_document" "default" {
+  statement {
+    sid = ""
+
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    effect = "Allow"
+  }
+}
+
+data "aws_ami" "default" {
```

Comment:
why use these `aws_ami` lookups in `main.tf`?
they will be different for every project.
it's ok to put it in the examples.

Reply:
I missed that :-P